protocol/verification.go

```diff
@@ -0,0 +1,56 @@
+package protocol
+
+import (
+	"bytes"
+
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	m "github.com/coniks-sys/coniks-go/merkletree"
+)
+
+func verifyAuthPath(uname string, key []byte,
+	ap *m.AuthenticationPath,
+	str *m.SignedTreeRoot) ErrorCode {
+
+	// verify VRF Index
+	vrfKey := vrf.PublicKey(str.Policies.VrfPublicKey)
+	if !vrfKey.Verify([]byte(uname), ap.LookupIndex, ap.VrfProof) {
+		return ErrorBadVRFProof
+	}
+
+	if bytes.Equal(ap.LookupIndex, ap.Leaf.Index) { /* proof of inclusion */
+		// verify name-to-key binding
+		if !ap.VerifyBinding(key) {
+			return ErrorBadBinding
+		}
+		// verify commitment
+		if !ap.Leaf.Commitment.Verify([]byte(uname), ap.Leaf.Value) {
+			return ErrorBadCommitment
+		}
+	}
+
+	// verify auth path
+	if !ap.Verify(str.TreeHash) {
+		return ErrorBadMapping
+	}
+
+	return Passed
+}
+
+func verifySTR(signKey sign.PublicKey,
+	str *m.SignedTreeRoot,
+	savedSTR []byte) ErrorCode {
+
+	// verify STR's signature
+	if !signKey.Verify(str.Serialize(), str.Signature) {
+		return ErrorBadSignature
+	}
+
+	// verify hash chain
+	if str.Epoch > 0 && savedSTR != nil &&
```

Comment:
Should we publish the signature of the STR at epoch `0` instead of publishing the seed? This way, we don't need to worry should the client verifies the seed or the signature at epoch `0`.